src/test/suite/checkFirstSuggestionDecoration.test.ts

```diff
@@ -0,0 +1,125 @@
+import * as assert from "assert";
+import { beforeEach, test, afterEach, suite } from "mocha";
+import * as sinon from "sinon";
+import { reset } from "ts-mockito";
+import { EventEmitter } from "events";
+import * as vscode from "vscode";
+import { activate, getDocUri, sleep } from "./utils/helper";
+import {
+  isProcessReadyForTest,
+  readLineMock,
+  requestResponseItems,
+  stdinMock,
+  stdoutMock,
+} from "../../binary/mockedRunProcess";
+import { resetBinaryForTesting } from "../../binary/requests/requests";
+import {
+  acceptInline,
+  makeAChange,
+  mockAutocomplete,
+  mockGetState,
+  moveToActivePosition,
+  triggerInline,
+} from "./utils/completion.utils";
+import {
+  anAutocompleteResponse,
+  INLINE_NEW_PREFIX,
+  INLINE_PREFIX,
+  SINGLE_CHANGE_CHARACTER,
+} from "./utils/testData";
+import * as vscodeUtils from "../../vscode.api";
+import { firstSuggestionDecorationType } from "../../firstSuggestionDecoration";
+
+const nonEmptyArray = sinon.match((v) => Array.isArray(v) && v.length);
+
+suite("First suggestion decoration", () => {
+  const docUri = getDocUri("firstSuggestion.txt");
+
+  beforeEach(async () => {
+    await activate(docUri);
+  });
+
+  afterEach(async () => {
+    reset(stdinMock);
+    reset(stdoutMock);
+    reset(readLineMock);
+    requestResponseItems.length = 0;
+    resetBinaryForTesting();
+    sinon.verifyAndRestore();
+    await vscode.commands.executeCommand("workbench.action.closeActiveEditor");
+  });
+
+  test("should display first suggestion on the first inline suggestion", async () => {
+    const events: EventEmitter = new EventEmitter();
+    const original = vscodeUtils.setDecoration;
+
+    const setDecorationStub = sinon
+      .stub(vscodeUtils, "setDecoration")
+      .callsFake((...args) => {
+        if (args[0] === firstSuggestionDecorationType) {
+          if (args[1].length === 0) {
+            events.emit("cleared");
+          } else {
+            events.emit("shown");
+          }
+        }
+        original(...args);
+      });
+    const decorationShown = new Promise((res) => events.once("shown", res));
```

Comment:
not sure I understand this, once is promise based already

Reply:
Node's `EventEmitter.once` is only a hook for a callback to do when an event happens. It's not promise based ([see here](https://github.com/nodejs/node/issues/20893)). [Emittery's `once`](https://www.npmjs.com/package/emittery) does return a promise though, but it requires adding a dependency.